### PR TITLE
Recast Navigation Gem docs and user guide

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>

## Change summary

Adds first documentation for the new Recast Navigation gem, the basics and the user guide for it. The user guide covers creation of a navigation mesh, enabling visualization and finding paths within a navigation mesh.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/gems/reference/physics/nvidia/recast-navigation.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+The Recast Navigation Gem provides support for building navigation meshes and calculating walkable paths within those navigation meshes. It uses open source library [Recast Navigation](https://github.com/recastnavigation/recastnavigation).
+
+An example of its use can be found in AutomatedTesting project in Navigation Sample level.
+
+This Gem includes the following features:
++ Select an area to build a navigation mesh from PhysX colliders
++ Create a navigation mesh
++ Configure navigation mesh parameters
++ Find paths between entities or positions within a navigation mesh
++ Visualize the navigation mesh
```
Comment (hultonha) [inline content/docs/user-guide/gems/reference/physics/nvidia/recast-navigation.md:17]: question: is `+` the preferred character for bulleted lists? I think it's generally preferred to use `-` but I'm not 100% - I definitely recommend installing Markdown Lint for VSCode if you're using it as it highlights a lot of these types of conventions

File: content/docs/user-guide/interactivity/navigation-and-pathfinding/recast-navigation.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+---
+linkTitle: Using Recast Navigation
+title: Finding Paths using Recast Navigation Gem in Open 3D Engine
+description: Instructions for setting up and configuring navigation mesh using Open 3D Engine (O3DE).
+---
+
+You can create a navigation mesh by using **Recast Navigation Mesh** component and its dependencies, and then use **Detour Navigation** component to find path between two entities or two points within the navigation area.
```
Comment (hultonha) [inline content/docs/user-guide/interactivity/navigation-and-pathfinding/recast-navigation.md:7]: nit: find path -> find a path

**Review by hultonha — APPROVED** [review]
This looks like a fantastic start to me @AMZN-Olex, great work! 🥳

File: content/docs/user-guide/interactivity/navigation-and-pathfinding/recast-navigation.md (hunk 2 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/interactivity/navigation-and-pathfinding/recast-navigation.md]: Done

**Comment by FiniteStateGit** [comment]
Hey @AMZN-Olex,

Could you create the `ai` and `recast` folders here: content/docs/user-guide/gems/reference/ai/recast.

We can move the gem reference `recast-navigation.md` out of physics/nvidia.  I'll fast-follow with an editorial review asap (I don't mind if we fix hyperlinks in the review).

File: content/docs/user-guide/gems/reference/physics/nvidia/recast-navigation.md (hunk 1 above)
Comment (hultonha) [reply content/docs/user-guide/gems/reference/physics/nvidia/recast-navigation.md]: Awesome thanks!

**Comment by AMZN-Olex** [comment]
> Hey @AMZN-Olex,
> 
> Could you create the `ai` and `recast` folders here: content/docs/user-guide/gems/reference/ai/recast.
> 
> We can move the gem reference `recast-navigation.md` out of physics/nvidia. I'll fast-follow with an editorial review asap (I don't mind if we fix hyperlinks in the review).

@FiniteStateGit Done!

**Pull request merged**